media-autobuild_suite.bat

```diff
@@ -1740,6 +1748,17 @@ title MABSbat
 if exist %build%\compilation_failed del %build%\compilation_failed
 if exist %build%\fail_comp del %build%\compilation_failed
 
+set nixdir=%instdir:\=/%
+set nixdir=%nixdir::=%
+
+echo.ffmpegPath: %ffmpegPath%
+
+if [%ffmpegPath:~0,2%]==[..] (
+	set ffmpegPath=/%nixdir%/%ffmpegPath%
+	)
+
+echo.ffmpegPath: %ffmpegPath%
+
```

Comment:
This whole logic might be better off being grouped in line 173-180 with a message of `Using <path> for FFmpeg` or something, rather than all the way down here 

Also, seems the two echos are redundant? Perhaps one was a debug print?

Reply:
You may also want to echo out directions/infomation while this is being set (so in the `if %ffmpegPathINI%==0` block).

So something like 

you would need to add it inside the media-autobuild_suite.ini file like
```ini
ffmpegPath=https://github.com/softworkz/FFmpeg.git#branch=submit_qsv_sei
```
or something, else document this inside the README.md

I at least think the above would work, haven't checked out this branch and tested it yet

Reply:
> This whole logic might be better off being grouped in line 173-180

For some obscure reason, the same commands don't work when placed there. 
I remembered this after I had tried to move it up there. I don't remember whether I ever figured out the reason. It's this statement

```
if [%ffmpegPath:~0,2%]==[..] (
    set ffmpegPath=/%nixdir%/%ffmpegPath%
)
```
which is always producing

`The syntax of the command is incorrect.`

But not when placed down in the range of 1700 .

Reply:
I know that batch has a lot of horrible quirks with variables, disqualifying it from being a sane scripting language, let me see if there's a way, and I'll get back to this

Reply:
Maybe it's something about setlocal and delayed expansion.  
At least we know now how the two echo statements came to be there :laughing:

Reply:
I came up with https://github.com/1480c1/media-autobuild_suite/commit/4ce390ce6fec0d531d63b9310054653b11268238

and it seems to work locally,

first run after switching to the branch, reusing an old build
```bat
.\media-autobuild_suite.bat
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------

 Using default ffmpeg source path: https://git.ffmpeg.org/ffmpeg.git

 If you want to use a custom source repository, add a line like this
 to media-autobuild_suite.ini:

     ffmpegPath=https://github.com/username/FFmpeg.git#branch=branchname

 or for a local repository like:

     ffmpegPath=../myrepos/ffmpeg

-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
update autobuild suite
-------------------------------------------------------------------------------
```

second run
```bat
.\media-autobuild_suite.bat
-------------------------------------------------------------------------------
update autobuild suite
-------------------------------------------------------------------------------
```

after modifying .\build\media-autobuild_suite.ini with `ffmpegPath=C:\Users\cddeg\FFmpeg`
```bat
.\media-autobuild_suite.bat
-------------------------------------------------------------------------------

 Using ffmpeg path: C:\Users\cddeg\FFmpeg

-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
update autobuild suite
-------------------------------------------------------------------------------
```

after using `ffmpegPath=../FFmpeg
```
.\media-autobuild_suite.bat
-------------------------------------------------------------------------------

 Using ffmpeg path: /D/FFmpeg

-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
update autobuild suite
-------------------------------------------------------------------------------
```

Reply:
That was made before your update, so you will need to adapt it as you see fit

Reply:
Besides the subroutine call, my update is identical.

For this...

```
set "resolvePath=%~dpnx1"
```

...I don't even want to know what it is and why it works.  :-)

Reply:
https://stackoverflow.com/a/18642270
> expands to the drive, path, name and extension of the first argument

Reply:
Will the result be relative to %instdir% when the script is called from a different dir?

Reply:
It will always be relative to instdir because of the cd at the top of the script

Reply:
Perfect!

Reply:
Updated with your changes included